l10n_it_corrispettivi_sale/tests/test_corrispettivi_sale.py

```diff
@@ -0,0 +1,124 @@
+            'corrispettivi': True
+        })
+        self.no_corr_fiscal_position = fiscal_position_model.create({
+            'name': 'corrispettivi fiscal position',
+            'corrispettivi': False
+        })
+        self.corrispettivi_partner = partner_model.create({
+            'name': 'Corrispettivi partner',
+            'use_corrispettivi': True,
+            'property_account_position_id': self.corr_fiscal_position.id
+        })
+        self.no_corrispettivi_partner = partner_model.create({
+            'name': 'Corrispettivi partner',
+            'use_corrispettivi': False,
+            'property_account_position_id': self.no_corr_fiscal_position.id
+        })
+
+        self.account_receivable = self.env['account.account'].search(
+            [('user_type_id', '=', self.env.ref(
+                'account.data_account_type_receivable').id)], limit=1)
+        self.product = self.env['product.product'].search([], limit=1)
+
+    def create_corr_journal(self):
+        corr_journal_id = self.env['account.journal'].create({
+            'name': 'CORR',
+            'code': 'CORR',
+            'type': 'sale',
+            'corrispettivi': True
+        })
+        return corr_journal_id
+
+    def create_no_corr_journal(self):
+        no_corr_journal_id = self.env['account.journal'].create({
+            'name': 'NOCORR',
+            'code': 'NOCORR',
+            'type': 'sale',
+            'corrispettivi': False
+        })
+        return no_corr_journal_id
+
+    def create_corrispettivi_sale(self):
+        corr_sale = self.sale_model \
+            .create({
+                'partner_id': self.corrispettivi_partner.id,
+                'order_line': [(0, 0, {
+                    'product_id': self.product.id
+                })]})
+        corr_sale.onchange_partner_corrispettivi_sale()
+        return corr_sale
+
+    def create_no_corrispettivi_sale(self):
+        no_corr_sale = self.sale_model \
+            .create({
+                'partner_id': self.no_corrispettivi_partner.id,
+                'order_line': [(0, 0, {
+                    'product_id': self.product.id
+                })]})
+        no_corr_sale.onchange_partner_corrispettivi_sale()
+        return no_corr_sale
+
+    def test_corrispettivi_sale_onchange(self):
+        """ Test onchange in sale order. """
+        sale = self.create_corrispettivi_sale()
+        self.assertTrue(sale.corrispettivi)
+
+        sale.partner_id = self.no_corrispettivi_partner
+        sale.onchange_partner_corrispettivi_sale()
+        self.assertFalse(sale.corrispettivi)
+
+    def test_corrispettivi_sale_invoice(self):
+        """
+        Test corrispettivo creation for a sale order having flag corrispettivi.
+        """
+        self.create_corr_journal()
+        sale = self.create_corrispettivi_sale()
+        self.assertTrue(sale.action_confirm())
+        invoice_ids_list = sale.action_invoice_create()
```

Comment:
Add check to be sure that invoice_ids_list is not empty

Reply:
Done in https://github.com/OCA/l10n-italy/pull/596/commits/93d7b51596d6ccc14ae933754eee6320f0cf23c3

Reply:
@eLBati I'll fixup when you confirm that it is ok for you https://github.com/OCA/l10n-italy/commit/93d7b51596d6ccc14ae933754eee6320f0cf23c3